upup/pkg/fi/cloudup/stage_inventory.go

```diff
@@ -0,0 +1,388 @@
+	outputValues := strings.Split(string(output), "Loaded image: ")
+	return strings.Trim(outputValues[1], "\n")
+}
+
+func downloadFile(url string, destPath string, dirMode os.FileMode) error {
+	err := os.MkdirAll(path.Dir(destPath), dirMode)
+	if err != nil {
+		return fmt.Errorf("error creating directories for destination file %q: %v", destPath, err)
+	}
+
+	output, err := os.Create(destPath)
+	if err != nil {
+		return fmt.Errorf("error creating file for download %q: %v", destPath, err)
+	}
+	defer output.Close()
+
+	glog.Infof("Downloading %q", url)
+
+	response, err := http.Get(url)
+	if err != nil {
+		return fmt.Errorf("error doing HTTP fetch of %q: %v", url, err)
+	}
+	defer response.Body.Close()
+
+	_, err = io.Copy(output, response.Body)
+	if err != nil {
+		return fmt.Errorf("error downloading HTTP content from %q: %v", url, err)
+	}
+	return nil
+}
+
+// Stolen from: https://play.golang.org/p/4FkNSiUDMg
+// newUUID generates a random UUID according to RFC 4122
+func NewUUID() (string, error) {
+	uuid := make([]byte, 16)
+	n, err := io.ReadFull(rand.Reader, uuid)
+	if n != len(uuid) || err != nil {
+		return "", err
+	}
+	// variant bits; see section 4.1.1
+	uuid[8] = uuid[8]&^0xc0 | 0x80
+	// version 4 (pseudo-random); see section 4.1.3
+	uuid[6] = uuid[6]&^0xf0 | 0x40
+	return fmt.Sprintf("%x-%x-%x-%x-%x", uuid[0:4], uuid[4:6], uuid[6:8], uuid[8:10], uuid[10:]), nil
+}
+
+func tagAndPushToDocker(tagName, imageId string) error {
+
+	glog.Infof("Tagging local image tagName[-]%s\n", tagName)
+	args := []string{"tag", imageId, tagName}
+	err := performExec(dockerExec, args)
+	if err != nil {
+		return fmt.Errorf("Docker Error - tagging tagName '%s' - imageId '%s' : %v", tagName, imageId, err)
+	}
+
+	// Push image to new Repo
+	glog.Infof("Pushing image tagName[-]%s\n", tagName)
+	args = []string{"push", tagName}
+	err = performExec(dockerExec, args)
+	if err != nil {
+		return fmt.Errorf("Docker Error - pushing tagName '%s': %v", tagName, err)
+	}
+
+	return nil
+}
+
+func cleanUpDockerImages(pushedImageId, baseImageId string) error {
+
+	args := []string{"rmi", "-f", pushedImageId}
+	glog.Infof("Removing pushed container image %s\n", strings.Join(args, " "))
+	err := performExec(dockerExec, args)
+	if err != nil {
+		return fmt.Errorf("Docker Error - removing pushed container image '%s': %v", pushedImageId, err)
+	}
+
+	args = []string{"rmi", "-f", baseImageId}
+	glog.Infof("Removing base container image %s\n", strings.Join(args, " "))
+	err = performExec(dockerExec, args)
+	if err != nil {
+		return fmt.Errorf("Docker Error - removing base container image '%s': %v", baseImageId, err)
+	}
+
+	return nil
+}
+
+func performExec(cmdStr string, args []string) error {
+
+	binary, err := exec.LookPath(cmdStr)
+	if err != nil {
+		return fmt.Errorf("Error finding executable file: %s - %v", cmdStr, err)
+	}
+
+	cmd := exec.Command(binary, args...)
+	var out bytes.Buffer
+	var errOut bytes.Buffer
+	cmd.Stdout = &out
```

Comment:
@robertojrojas can you take this as a TODO please.  Also, do we have exec code in the codebase? I would be surprised if we do not have a util method.